File: src/main/java/com/hotel/entity/Notification.java
package com.hotel.entity;

import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "notifications", indexes = {
    @Index(name = "idx_user_unread", columnList = "user_id, is_read"),
    @Index(name = "idx_user_created", columnList = "user_id, created_at")
})
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 50, name = "user_id")
    private String userId;

    @Column(nullable = false, length = 100)
    private String title;

    @Column(nullable = false, length = 500)
    private String message;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    private NotificationType type;

    @Column(nullable = false, name = "is_read")
    private Boolean isRead = false;

    @Column(nullable = false)
    private Integer priority = 0;

    @Column(length = 200)
    private String actionLink;

    @CreationTimestamp
    @Column(nullable = false, name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "read_at")
    private LocalDateTime readAt;

    @PrePersist
    protected void onCreate() {
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
        if (isRead == null) {
            isRead = false;
        }
        if (priority == null) {
            priority = 0;
        }
    }
}

File: frontend/src/types/notification.ts
export type NotificationType = 'BOOKING' | 'CANCELLATION' | 'CHECK_IN' | 'CHECK_OUT' | 'SYSTEM'

export interface Notification {
  id: number
  title: string
  message: string
  type: NotificationType
  isRead: boolean
  priority: number
  actionLink: string | null
  createdAt: string
  readAt: string | null
}

export interface NotificationListResponse {
  notifications: Notification[]
  total: number
  unreadCount: number
}

export interface ChangePasswordRequest {
  oldPassword: string
  newPassword: string
}
